File: app/src/main/java/com/example/recyclerview/QuoteViewHolder.java
package com.example.recyclerview;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

public class QuoteViewHolder extends ViewHolder<String>
{
    private final TextView quoteTextView = (TextView) getView().findViewById(R.id.quote);

    private QuoteViewHolder(View itemView)
    {
        super(itemView);
    }

    public static QuoteViewHolder make(ViewGroup parent)
    {
        LayoutInflater viewInflater = LayoutInflater.from(parent.getContext());
        View quoteListItemView = viewInflater.inflate(R.layout.listitem_quote,parent,false);
        return new QuoteViewHolder(quoteListItemView);
    }

    @Override
    protected void onSetModel(String newModel)
    {
        quoteTextView.setText(newModel);
        quoteTextView.setBackgroundColor(newModel.hashCode());
    }
}

File: app/src/main/java/com/example/recyclerview/QuoteViewHolder.kt
package com.example.recyclerview

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView


class QuoteViewHolder private constructor(view:View):ViewHolder<String>(view)
{
    companion object
    {
        fun new(parent:ViewGroup):QuoteViewHolder
        {
            val viewInflater = LayoutInflater.from(parent.context)
            val quoteListItemView = viewInflater.inflate(R.layout.listitem_quote,parent,false)
            return QuoteViewHolder(quoteListItemView)
        }
    }

    private val quoteTextView = view.findViewById(R.id.quote) as TextView

    override fun onSetModel(model:String)
    {
        quoteTextView.text = model
        quoteTextView.setBackgroundColor(model.hashCode())
    }
}
